sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/promotion/SyncGetOnOtherPlatformsActivity.kt

```diff
@@ -0,0 +1,104 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.sync.impl.promotion
+
+import android.annotation.SuppressLint
+import android.content.ActivityNotFoundException
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.flowWithLifecycle
+import androidx.lifecycle.lifecycleScope
+import com.duckduckgo.anvil.annotations.InjectWith
+import com.duckduckgo.common.ui.DuckDuckGoActivity
+import com.duckduckgo.common.ui.viewbinding.viewBinding
+import com.duckduckgo.di.scopes.ActivityScope
+import com.duckduckgo.navigation.api.GlobalActivityStarter
+import com.duckduckgo.sync.impl.R
+import com.duckduckgo.sync.impl.databinding.ActivitySyncGetOnOtherDevicesBinding
+import com.duckduckgo.sync.impl.promotion.SyncGetOnOtherPlatformsViewModel.Command
+import com.duckduckgo.sync.impl.promotion.SyncGetOnOtherPlatformsViewModel.Command.ShareLink
+import com.duckduckgo.sync.impl.promotion.SyncGetOnOtherPlatformsViewModel.Command.ShowCopiedNotification
+import com.google.android.material.snackbar.Snackbar
+import javax.inject.Inject
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import timber.log.Timber
+
+@InjectWith(ActivityScope::class)
+class SyncGetOnOtherPlatformsActivity : DuckDuckGoActivity() {
+
+    private val viewModel: SyncGetOnOtherPlatformsViewModel by bindViewModel()
+    private val binding: ActivitySyncGetOnOtherDevicesBinding by viewBinding()
+
+    @Inject
+    lateinit var globalActivityStarter: GlobalActivityStarter
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        viewModel.commands.flowWithLifecycle(lifecycle, Lifecycle.State.STARTED)
+            .onEach { executeCommand(it) }
+            .launchIn(lifecycleScope)
+
+        setContentView(binding.root)
+        setupToolbar(binding.includeToolbar.toolbar)
+        configureUiEventHandlers()
+    }
+
+    private fun configureUiEventHandlers() {
+        binding.shareLinkButton.setOnClickListener {
+            viewModel.onShareClicked()
+        }
+        binding.downloadLinkText.setOnClickListener {
+            viewModel.onLinkClicked()
+        }
+    }
+
+    private fun executeCommand(command: Command) {
+        when (command) {
+            is ShareLink -> launchSharePageChooser(command.link)
+            is ShowCopiedNotification -> showCopiedNotification()
+        }
+    }
+
+    private fun showCopiedNotification() {
+        Snackbar.make(binding.root, R.string.syncGetAppsOnOtherPlatformInstructionUrlLinkCopied, Snackbar.LENGTH_SHORT).show()
+    }
+
+    @SuppressLint("UnspecifiedImmutableFlag")
+    private fun launchSharePageChooser(link: String) {
+        val share = Intent(Intent.ACTION_SEND).apply {
+            type = "text/plain"
+            putExtra(Intent.EXTRA_TEXT, link)
```

Comment:
@karlenDimla, just a heads up that this was changed since you reviewed. Previously the link contained a surrounding piece of instructions and we've stripped it back to just sharing a link.

Just FYI for transparency.